src/bTagSF.cc

```diff
@@ -129,6 +139,50 @@ float bTagSF::getSF (WP wpt, SFsyst syst, int jetFlavor, float pt, float eta)
     return SF;
 }
 
+float bTagSF::getSFshifted (std::string systName, int jetFlavor, float pt, float eta)
+{
+    float SF = 1.0;
+
+    BTagEntry::JetFlavor flav;
+    float mypt = pt;
+    if (mypt < 20.) mypt = 20.;
+    if (mypt > 1000.) mypt = 1000.;
+
+    if (abs(jetFlavor) == 5)
+    {
+        flav = BTagEntry::FLAV_B;
+    }
+    else if (abs(jetFlavor) == 4)
+    {
+        flav = BTagEntry::FLAV_C;
+    }
+    else
+    {
+        flav = BTagEntry::FLAV_UDSG;
+    }
+
+    if (DEBUG) cout << "   ~~ requesting SF for systName= " << systName << " BTagEntry::FLAV=" << flav << " pt=" << pt << " eta=" << eta << endl;
+
+    // From Twiki: Owing to the iterative approach, "jes", "lf", "hf", "hfstats1/2",
+    // and "lfstats1/2" uncertainties are applied to both b and udsg jets.
+    // For c-flavored jets, only "cferr1/2" uncertainties are applied.
+    if (flav != BTagEntry::FLAV_C && systName.find("cferr") != std::string::npos)
+    {
+        SF = 1.;
+    }
+    else if (flav == BTagEntry::FLAV_C && !(systName.find("cferr") != std::string::npos) )
```

Comment:
I would put (flav == BTagEntry::FLAV_C && systName.find("cferr") == std::string::npos )

Reply:
fixed in 2e10aea